modules/uhttp/router.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package uhttp
+
+import (
+	"github.com/gorilla/mux"
+
+	"go.uber.org/fx/service"
+)
+
+type route struct {
+	handler Handler
+}
+
+// Router is wrapper around gorila mux
+type Router struct {
+	mux.Router
+
+	host service.Host
+}
+
+// NewRouter creates a new empty router
+func NewRouter(host service.Host) *Router {
+	return &Router{
+		host: host,
+	}
+}
+
+// HandleRoute wraps and calls the http.Handler underneath
+func (h *Router) HandleRoute(host service.Host, path string, handler Handler) {
```

Comment:
Why not just call this `Handle` and then change the next line to `h.Router.Handle()`?